src/Flex.js

```diff
@@ -1,13 +1,23 @@
-import {withSystemProps, FLEX_CONTAINER, COMMON, FLEX_ITEM} from './system-props'
+import styled from 'styled-components'
+import {display} from 'styled-system'
+import theme from './theme'
+import {FLEX_CONTAINER, FLEX_ITEM, COMMON} from './constants'
 
-const Flex = withSystemProps(
-  {
-    is: 'div',
-    display: 'flex'
-  },
-  [...FLEX_CONTAINER, ...COMMON]
-)
+const Flex = styled.div`
+  ${FLEX_CONTAINER}
+  ${COMMON}
+  ${display}
+`
```

Comment:
@colebemis @shawnbot  hmm yeah honestly the only reason I left the syntax like that was because I'd previously had some CSS in there 😂 I like having consistent syntax but if there's a performance penalty then I don't think doing `const Flex = styled.div(FLEX_CONTAINER, COMMON, display)` is too confusing of a departure from the tagged template literal syntax

Reply:
Chiming in here 👋 since I know a little bit about styled-components 🙈: our Babel plugin automatically optimizes tagged template literals under the hood for perf. For example, by default babel-preset-env (or whatever the relevant Babel plugin is that's used under the hood) would turn this component:

```JS
const Simple = styled.div`
  width: 100%;
`
```

into this JS:

```JS
var _templateObject = _taggedTemplateLiteral(['width: 100%;'], ['width: 100%;'])
function _taggedTemplateLiteral(strings, raw) {
  return Object.freeze(Object.defineProperties(strings, { raw: { value: Object.freeze(raw) } }))
}
var Simple = _styledComponents2.default.div(_templateObject)
```

which is done to be 100% spec-compliant and most of which is 100% useless when writing styled-components.

Our Babel plugin pre-empts that transformation and instead turns it into this much faster and smaller version ✨:

```JS
var Simple = _styledComponents2.default.div(['width: 100%;'])
```

You can read more about that transformation [in the docs](https://www.styled-components.com/docs/tooling#template-string-transpilation), but TL;DR: use the Babel plugin it's pretty good 😊